etcdserver/api/membership/cluster.go

```diff
@@ -574,6 +575,9 @@ func (c *RaftCluster) SetVersion(ver *semver.Version, onSet func(*zap.Logger, *s
 		mustSaveClusterVersionToBackend(c.be, ver)
 	}
 	ClusterVersionMetrics.With(prometheus.Labels{"cluster_version": ver.String()}).Set(1)
+	if oldVer != nil {
+		ClusterVersionMetrics.With(prometheus.Labels{"cluster_version": oldVer.String()}).Set(0)
```

Comment:
Should we set the old version first? And set the latest version afterwards. The later cluster version may be overridden, if two are the same.

Reply:
Good catch! Thanks!

Today we only set cluster version if the new version is strictly higher than the old. But this will not be true once we start to support HA downgrade.

Reply:
Fixed.

Reply:
etcd_cluster_version{cluster_version="3.0"} 0
etcd_cluster_version{cluster_version="3.4"} 1

set 1 means this is 3.4 version up. set 0 means 3.0 version did off? @jingyih